src/dev/so_migration/snapshot_plugin_types.ts

```diff
@@ -92,7 +92,7 @@ function collectSOSnapshotMeta(): MigrationSnapshotMeta {
   const buildUrl = process.env.BUILDKITE_BUILD_URL || null;
   const prId = process.env.BUILDKITE_MESSAGE?.match(/\(#(\d+)\)/)?.[1];
   const pullRequestUrl = prId ? `https://github.com/elastic/kibana/pulls/${prId}` : null;
-  const kibanaCommitHash = process.env.BUILDKITE_COMMIT || getLocalHash();
+  const kibanaCommitHash = process.env.BUILDKITE_COMMIT || getLocalHash() || 'unknown';
```

Comment:
The type changed below as well, why not use `null`?

Reply:
I wanted to avoid having to use `!` on the usage sites of the de-serialized objects. It's either tightening this type, or using an "I'm telling you!" operator, and I thought, realistically, this will always be available (as the `BUILDKITE_COMMIT` is available on CI, and the local hash is available locally), so let's reflect that in the types.